src/Pack/Core/Logging.idr

```diff
@@ -1,11 +1,55 @@
 module Pack.Core.Logging
 
+import Control.Monad.Either
+import Control.Monad.Trans
 import Pack.Core.Types
 
 import System
 
 %default total
 
+--------------------------------------------------------------------------------
+--          Support
+--------------------------------------------------------------------------------
+
+public export
+data ConfirmResult = Yes | No | Unknown
+
+||| Converts a string into a `ConfirmResult`, ignoring case.
+||| Recognises `"y"` or `"yes"` as `Yes`, `"n"` or `"no"` as `No`.
+||| An empty string defaults to `No`. Any other input is treated as `Unknown`.
+parseConfirmResult : String -> ConfirmResult
+parseConfirmResult x = case toLower x of
+  "y"   => Yes
+  "yes" => Yes
+  "n"   => No
+  "no"  => No
+  ""    => No -- Default
+  _     => Unknown
+
+||| Constructs a prompt message for a confirmation dialog.
+confirmMessage : String -> String
+confirmMessage msg =
+  if isLong
+     then "\{msg}\n\n\{continueMessage}"
+     else "\{msg} \{continueMessage}"
+  where
+    isLong : Bool
+    isLong = length msg > 80 || isInfixOf "\n\n" msg
+
+    continueMessage : String
+    continueMessage = "Continue (yes/*no)?"
+
+||| Executes an action that returns a `ConfirmResult` and ensures
+||| that it is confirmed by the user. If not, throws `SafetyAbort`.
+mustConfirm :
+     HasIO io
+  => io ConfirmResult
```

Comment:
Please don't you double arrows in multi-line type signatures. Use `{auto ...}` instead, for better readability.